### DISCUSSION TITLE
GitHub actions cannot find Gradle/maven packages

### DISCUSSION THREAD

**Discussion by rafal-jambit** [community/community#56824] [Actions] [answered]
### Select Topic Area

Question

### Body

We are working with Gradle (8.1.1 as of this writing).
We have a private repository A that is also hosting 2 packages (Android AARs, a debug and release versions of the same library). These packages have been uploaded with `./gradlew publish` manually (for now) from within repository A and they upload just fine, we can see them in the GitHub web UI.
Then, in private repository B we add these packages from A as dependencies. For this, we configure the maven repository with a URL containing repository A in the URL, and the necessary auth information, as references to env variables:
```
allprojects {
    repositories {
        ...
        maven {
            url 'https://maven.pkg.github.com/customer/rclib'
            credentials {
                username System.getenv('CUSTOMER_GITHUB_USERNAME')
                // A GitHub personal token ('classic') with 'read:packages' scope.
                password System.getenv('CUSTOMER_GITHUB_PASSWORD')
            }
        }
    }
}
```
(The necessary envs are in my shell configuration files.)
When we build B locally, it all works fine, the dependencies are downloaded and the project builds and runs fine. Authentication info is: the username is my GitHub username and the password is a classic personal token with `write:packages` (I used the same one to upload the packages in the first place).

When we try to build project B in a workflow hosted in '<repo B>/.github/workflows/ci.yaml', though, it always fails with the following error (redacted slightly):
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':rclib:generateReleaseRFile'.
> Could not resolve all files for configuration ':rclib:releaseCompileClasspath'.
   > Failed to transform rclib-customer-release-2.6.9.aar (com.customer:rclib-customer-release:2.6.9) to match attributes {artifactType=android-symbol-with-package-name, org.gradle.category=library, org.gradle.dependency.bundling=external, org.gradle.libraryelements=aar, org.gradle.status=release, org.gradle.usage=java-api}.
      > Could not find rclib-customer-release-2.6.9.aar (com.customer:rclib-customer-release:2.6.9).
        Searched in the following locations:
            https://maven.pkg.github.com/customer/rclib/com/customer/rclib-customer-release/2.6.9/rclib-customer-release-2.6.9.aar
```
However, when I click the link and I authenticate using basic auth with my credentials (the user and personal classic token mentioned above) I can download the files just fine, which means they are really there.

I initially used `GITHUB_ACTOR` and `GITHUB_TOKEN` as auth info in the workflow to set the env variables that our `build.gradle` requires, like this:
```
    - name: Set maven repository credentials
      run: |
        echo "CUSTOMER_GITHUB_USERNAME=$GITHUB_ACTOR" >> "$GITHUB_ENV"
        echo "CUSTOMER_GITHUB_PASSWORD=${{ secrets.GITHUB_TOKEN }}" >> "$GITHUB_ENV"
    - name: Build Android
      run: flutter build apk # eventually calls ./gradlew assembleRelease
```

I thought that maybe the token has too few privileges (although according to the docs and the output of the 'set up job' step it does have `read:packages`) and hardcoded my credentials in the pipeline instead (the ones that work locally), and I got exactly the same error. I even changed the code so that the pipeline doesn't set the env variables for credentials at all, and they are instead hardcoded in `build.gradle`, but I get still the same error. Somehow, the GitHub action can't find the files, and the credentials seem to have nothing to do with it.

What am I doing wrong?

**Comment by devmrcs** [comment]
I'm going through the same problem

Did you get a solution?

**Reply by rafal-jambit** [reply]
Not yet, since I wrote it my focus moved elsewhere, and I will come back to this issue this or next week.

**Reply by rafal-jambit** [reply, marked as answer]
I took a look into this today and ... it works. No idea what was wrong before, but it works now. We are using the personal access token (classic) for this, and here is what I'm doing:
```yaml
    - name: Set up GitHub Packages secrets
      run: |
        echo "GITHUB_PACKAGES_USERNAME=${{ secrets.PACKAGES_USERNAME }}" >> "$GITHUB_ENV"
        echo "GITHUB_PACKAGES_PASSWORD=${{ secrets.PACKAGES_PASSWORD }}" >> "$GITHUB_ENV"
```
Following steps have the credentials. In my Gradle file I define the repository as this:
```groovy
maven {
    url 'https://maven.pkg.github.com/my-org/my-repo'
    credentials {
        username = System.getenv('GITHUB_PACKAGES_USERNAME')
        // A GitHub personal token ('classic') with 'read:packages' scope.
        password = System.getenv('GITHUB_PACKAGES_PASSWORD')
    }
}
```
The repository with the package is in reality a different repository than the one the pipelines runs for.

Remembering that it didn't work before I tried to break it in all ways that came to my mind (even nuked the whole ~/.gradle to make sure no 'cached' deps are used), but it works fine.
